src/solver/application.cpp

```diff
@@ -0,0 +1,466 @@
+    logs.checkpoint() << "Antares Solver v" << ANTARES_VERSION_PUB_STR;
+    WriteHostInfoIntoLogs();
+    logs.info();
+
+#ifdef ANTARES_SWAP_SUPPORT
+    // Changing the swap folder
+    if (!pSettings.swap.empty())
+    {
+        logs.info() << "  memory pool: scratch folder:" << pSettings.swap;
+        Antares::memory.cacheFolder(pSettings.swap);
+    }
+    else
+        logs.info() << "  memory pool: scratch folder:" << Antares::memory.cacheFolder();
+#endif
+
+    // Initialize the main structures for the simulation
+    // Logs
+    Resources::WriteRootFolderToLogs();
+    logs.info() << "  :: log filename: " << logs.logfile();
+    // Temporary use a callback to count the number of errors and warnings
+    logs.callback.connect(this, &Application::onLogMessage);
+
+    // Allocate a study
+    pStudy = new Data::Study(true /* for the solver */);
+
+    // Setting global variables for backward compatibility
+    Data::Study::Current::Set(pStudy);
+    pParameters = &(pStudy->parameters);
+
+    // Loading the study
+    readDataForTheStudy(options);
+
+    // LISTE DE CHECKS ...
+
+    // CHECK incompatible de choix simultané des options « simplex range= daily » et « hydro-pricing
+    // = MILP ».
+    if ((pParameters->simplexOptimizationRange == Antares::Data::SimplexOptimization::sorDay)
+        && (pParameters->hydroPricing.hpMode == Antares::Data::HydroPricingMode::hpMILP))
+    {
+        throw Error::IncompatibleOptRangeHydroPricing();
+    }
+
+    // CHECK incompatible de choix simultané des options « simplex range= daily » et «
+    // unit-commitment = MILP ».
+    if ((pParameters->simplexOptimizationRange == Antares::Data::SimplexOptimization::sorDay)
+        && (pParameters->unitCommitment.ucMode == Antares::Data::UnitCommitmentMode::ucMILP))
+    {
+        throw Error::IncompatibleOptRangeUCMode();
+    }
+
+    // CHECK
+    // Daily simplex optimisation and any area's use heurictic target turned to "No" are not
+    // compatible.
+    if (pParameters->simplexOptimizationRange == Antares::Data::SimplexOptimization::sorDay)
+    {
+        for (uint i = 0; i < pStudy->areas.size(); ++i)
+        {
+            const auto& area = *(pStudy->areas.byIndex[i]);
+            if (!area.hydro.useHeuristicTarget)
+            {
+                throw Error::IncompatibleDailyOptHeuristicForArea(area.name);
+            }
+        }
+    }
+
+    // CHECK MinStablePower
+    bool tsGenThermal = (0
+                         != (pStudy->parameters.timeSeriesToGenerate
+                             & Antares::Data::TimeSeries::timeSeriesThermal));
+    logs.debug() << "tsGenThermal = " << tsGenThermal;
+
+    if (tsGenThermal)
+    {
+        std::map<int, YString> areaClusterNames;
+        if (!(pStudy->areasThermalClustersMinStablePowerValidity(areaClusterNames)))
+        {
+            Error::ThermalClusterHelper helper;
+            for (const auto& it : areaClusterNames)
+                helper.append(it.second);
```

Comment:
Remove append, helper. Provide list of clusters to `InvalidParametersForThermalClusters`